qiime2/core/path.py

```diff
@@ -27,14 +27,22 @@ def __new__(cls, *args, **kwargs):
         self._user_owned = True
         return self
 
+    def _copy_dir_or_file(self, other):
+        if self.is_dir():
+            return distutils.dir_util.copy_tree(str(self), str(other))
+        else:
+            return shutil.copy(str(self), str(other))
+
     def _move_or_copy(self, other):
         if self._user_owned:
-            if self.is_dir():
-                return distutils.dir_util.copy_tree(str(self), str(other))
-            else:
-                return shutil.copy(str(self), str(other))
+            return self._copy_dir_or_file(other)
         else:
-            return _ConcretePath.rename(self, other)
+            # Certain networked filesystems will experience a race
+            # condition on `rename`, so fall back to copying.
+            try:
+                return _ConcretePath.rename(self, other)
+            except FileExistsError:
+                return self._copy_dir_or_file(other)
```

Comment:
I think this will leave dangling files, as `self` will not be managing the lifetime of `other` as it would have originally.

Reply:
Are you specifically worried about the copy operation in this branch (not owned) - or in general (in both the owned and not owned branches)?

Reply:
only the un-owned branch. Basically, if a rename was good enough before, then we weren't planning on keeping the original path around (and so nothing will expect to go back to clean it up).

Reply:
Perfect, thanks for confirming!